addons/l10n_id_efaktur/models/account_move.py

```diff
@@ -129,7 +137,10 @@ def download_efaktur(self):
                 raise ValidationError(_('Could not download E-faktur in draft state'))
 
             if record.partner_id.l10n_id_pkp and not record.l10n_id_tax_number:
-                raise ValidationError(_('Connect %(move_number)s with E-faktur to download this report', move_number=record.name))
+                if not self.l10n_id_need_kode_transaksi:
+                    raise ValidationError(_('E-faktur is not available for invoices without any taxes.'))
+                else:
+                    raise ValidationError(_('Connect %(move_number)s with E-faktur to download this report', move_number=record.name))
```

Comment:
I would do something like this personally but it's just nit picking
```suggestion
                if not self.l10n_id_need_kode_transaksi:
                    raise ValidationError(_('E-faktur is not available for invoices without any taxes.'))
                raise ValidationError(_('Connect %(move_number)s with E-faktur to download this report', move_number=record.name))
```

Reply:
I do not have any preferences so I'll change to yours